### PR TITLE
[DOC] Update docs for 24.06.0 release [skip ci]

### PR DESCRIPTION
**Pull request by SurajAralihalli**
This PR updates download docs for 24.06.0 release.

Signed-off-by: Suraj Aralihalli [suraj.ara16@gmail.com](mailto:suraj.ara16@gmail.com)

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -83,23 +83,16 @@ CUDA 11.8 through CUDA 12.0.
...
 ### Release Notes
-* New functionality and performance improvements for this release include:
-* Performance improvements for S3 reading. 
-Refer to perfio.s3.enabled in [advanced_configs](./additional-functionality/advanced_configs.md) for more details.
-* Performance improvements when doing a joins on unique keys.
-* Enhanced decompression kernels for zstd and snappy.
-* Enhanced Parquet reading performance with modular kernels.
-* Added compatibility with Spark version 3.5.1.
-* Deprecated support for Databricks 10.4 ML LTS.
+* Improve support for Unity Catalog on Databricks.
```
Comment (SurajAralihalli) [inline docs/download.md:95]: @viadea @sameerz @mattahrens 

Please provide other suggestions to update release notes.

File: docs/download.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
 that uses the RAPIDS Accelerator For Apache Spark. See the [getting-started
 guide](https://docs.nvidia.com/spark-rapids/user-guide/latest/getting-started/overview.html) for more details.
 
-## Release v24.04.1
+## Release v24.06.0
```
Comment (NvTimLiu) [inline docs/download.md:21]: We support 3.4.3 spark shim from v24.06.0

**Review by NvTimLiu — APPROVED** [review]
LGTM, +1

**Comment by NvTimLiu** [comment]
build

**Comment by NvTimLiu** [comment]
build

**Review by sameerz — APPROVED** [review]

**Pull request merged**